src/main/java/com/solvd/app/utils/DOMParser.java

```diff
@@ -0,0 +1,92 @@
+package com.solvd.app.utils;
+
+import com.solvd.app.models.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.*;
+
+import javax.xml.parsers.*;
+import java.util.function.Function;
+
+public class DOMParser {
+
+    private static final Logger LOGGER = LogManager.getLogger(DOMParser.class);
+
+    public static void parseXMLFile(String xmlFilePath) {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+
+        try {
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document doc = builder.parse(xmlFilePath);
+
+            parseElements(doc, "pharmacy", DOMParser::parsePharmacy);
+            parseElements(doc, "person", DOMParser::parsePerson);
+            parseElements(doc, "position", DOMParser::parsePosition);
+            parseElements(doc, "staff", DOMParser::parseStaff);
+            parseElements(doc, "customer", DOMParser::parseCustomer);
+        } catch (Exception e) {
```

Comment:
Very general exception. Could use more description exceptions like ParserConfigurationException | SAXException | IOException

Reply:
Thanks for the suggestion. I just added more exceptions.